### PR TITLE
Build for 3.9.0

### PR DESCRIPTION
**Pull request by h-vetinari**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Version 3.9.0 has been released for a while, let's try building for it.

@conda-forge-admin, please rerender

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,8 +9,8 @@ package:
   version: "{{ version }}"
 
 source:
-  url: http://www.netlib.org/lapack/lapack-{{ version }}.tar.gz
-  sha256: deb22cc4a6120bff72621155a9917f485f96ef8319ac074a7afbc68aab88bcf6
```
Comment (h-vetinari) [inline recipe/meta.yaml]: The url for the sources [changed](http://www.netlib.org/lapack/#_lapack_version_3_9_0_2) between 3.8.0 and 3.9.0

File: recipe/meta.yaml [hunk 2]
```diff
@@ -179,9 +179,10 @@ outputs:
         - cmake
         - m2-make   # [win]
         - posix     # [win]
+        # hack (around which bug?)
```
Comment (h-vetinari) [inline recipe/meta.yaml]: This was noted as a hack in https://github.com/conda-forge/blas-feedstock/commit/a5df20db6d59e62a5bf94400e0cdadf4ea121396, can we document what's necessary to remove it?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -179,9 +179,10 @@ outputs:
...
         - llvm-openmp   # [linux and blas_impl == "openblas"]
       host:
-      # Building with blis fails due to a conda-build bug
+      # Building with blis fails due to a conda-build bug - which one?
```
Comment (h-vetinari) [inline recipe/meta.yaml]: Same here - which conda-build bug is relevant here? What would be necessary to simply use:
```
      host:
        - {{ pin_subpackage("libblas", exact=True) }}
        - {{ pin_subpackage("libcblas", exact=True) }}
      {% if blas_impl != 'blis' %}
        - {{ pin_subpackage("liblapack", exact=True) }}
        - {{ pin_subpackage("liblapacke", exact=True) }}
      {% endif %}
```

**Comment by h-vetinari** [comment]
OK, clearly I misunderstood this recipe, and how it's necessary to build https://github.com/conda-forge/lapack-feedstock for 3.9.0 first.

**Comment by h-vetinari** [comment]
I rebased this in anticipation of https://github.com/conda-forge/lapack-feedstock/pull/32 hopefully being finished soon.

I have to say that I really had trouble wrapping my head around this recipe. I couldn't understand (from the `meta.yaml`) how `libblas` depends on the netlib implementation:
```
PackagesNotFoundError: The following packages are not available from current channels:

  - libcblas==3.9.0[build=*netlib]
  - liblapacke==3.9.0[build=*netlib]
  - liblapack==3.9.0[build=*netlib]
  - libblas==3.9.0[build=*netlib]
```

It took me a while to find out that that dependency only appears in [`build.sh`](https://github.com/conda-forge/blas-feedstock/blob/master/recipe/build.sh#L24-L28) (which also gets called in [`bld.bat`](https://github.com/conda-forge/blas-feedstock/blob/master/recipe/bld.bat#L15), but curiously doesn't error the build upon failing).

Also, why do we download the netlib-sources again if the build is already done in the feedstock for [lapack](https://github.com/conda-forge/lapack-feedstock)? I'm thinking this could/should be made more explicit, no? Will try for a patch...

**Comment by h-vetinari** [comment]
OK, I've had a few pennies drop (see edits above), but now that I was trying to put this into words, I still have to wonder:

Why do we link against netlib at all if `build_pkg.(sh|bat)` will immediately replace it? Yes, there is a testrun of the netlib-packages being performed, by why not do that (including [`more_testing.diff`](https://github.com/conda-forge/blas-feedstock/blob/master/recipe/more_testing.diff)) in the lapack-feedstock?

What am I (still) missing @isuruf?

**Comment by h-vetinari** [comment]
On another note: After some tries, https://github.com/conda-forge/blas-feedstock/pull/43/commits/6b4fbef827dd30e43389588b69e811491877cc54 now correctly aborts windows builds on errors within `build.sh` as well.

**Comment by isuruf** [comment]
For testing, what we do here is, we build the test-suite of netlib and then we replace the implementation with `openblas` or `blis` or `mkl` and run the test suite to see if changing the blas implementation under the hood actually does work.

**Comment by h-vetinari** [comment]
> For testing, what we do here is, we build the test-suite of netlib and then we replace the implementation with `openblas` or `blis` or `mkl` and run the test suite to see if changing the blas implementation under the hood actually does work.

Understood, my point was that the test suite for the netlib implementation is already run in the lapack-feestock, and just running the tests here with `openblas` or `blis` or `mkl` should be enough, no?

**Comment by isuruf** [comment]
We are running the netlib test suite with openblas, etc. (We aren't running netlib test suite with netlib) This is done to check that these implementations are indeed compatible.

**Comment by h-vetinari** [comment]
OK, that makes sense! Thanks for the explanation.

**Comment by h-vetinari** [comment]
@isuruf
I tried to put what I understood in an explanatory comment (see https://github.com/conda-forge/blas-feedstock/pull/43/commits/316649ac5922f21fe30a478c714c667b66c2bba5). Happy to iterate if I'm still not getting something, but I'd hope that this can be included so that the next person looking at this recipe might be less lost than I am. ;-)

**Comment by h-vetinari** [comment]
Superseded by #52.

**Pull request closed without merging**